### DISCUSSION TITLE
Welcome to webtau Discussions!

### DISCUSSION THREAD

**Discussion by MykolaGolubyev** [testingisdocumenting/webtau#1005] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by aplatypus** [comment]
* Hope this is the right place to do this...
* Can I use either 
   *  Edge 
   * Firefox 

  ... with  WebTau?
  
*   If so which environment variables do I need to set?  I've seen:
   *  `WEBTAU_CHROME_DRIVER_PATH`
   * Which file it that looking for?  The chrome executable or something different?

I should addd that I'm running this from a Gradle project so I really need to know what's required -- It can be  _almost_ like running in a sandbox.

**Comment by MykolaGolubyev** [comment]
Hi @aplatypus,

To run your tests using Firefox (edge is not supported at the moment) you need to pass `browserId` to the runner. In case of groovy CLI runner `--browserId=firefox`, in case of JVM Junit like runners `-DbrowserId=firefox` or you can set env var `WEBTAU_BROWSER_ID=firefox`. 
Alternatively you can use `webtau.properties` file in resources and put `browserId=firefox` inside, but then it may become harder to modify if you need to switch back and forth. 

Webtau tries to download driver automatically. If you are behind a firewall, you can set location of the driver using `firefoxDriverPath` config value (system prop, or env var, following the same rule of name change)

As a side note, Gradle specific plugin to execute Groovy files is in a backlog, but I have little expertise implementing Gradle plugins as part of Maven project (webtau is maven based)

**Reply by aplatypus** [reply]
* I am using JUnit5 to do tests
* Our automated tests are run with JUnit by Gradle as part of the Github commit checks
* In a way you have a lot of infrastructure for 'free' via JUnit 5 and maven or gradle 
   * may be you can go that way and put time into the core aspects (???)
   * JUnit 5 seems to do well on Groovy.  Before I was using Spock with JUnit4 (spock is in Groovy)
* I would definitely recommend Gradle over Maven.  There's always that learning curve though ;-)
* Would the CHROME driver work with Chromumium?

**Comment by MykolaGolubyev** [comment]
Could you please clarify support of JUnit5? Webtau API is runner agnostic. It does also have dedicated JUnit5 support in terms of producing reports based on Junit5 test structure. 

Problem with Gradle in my case is webtau project itself is maven and I wasn't able to find a good way to build/register gradle plugins from maven project. I don't have plans to switch webtau to gradle itself. But may need to google more up-to-date info on how to release gradle plugins from maven. 

> Would the CHROME driver work with Chromumium?
should be able to support
